### ISSUE TITLE
`-webkit-fill-available`

### ISSUE THREAD

**Issue by bramus** [web-platform-tests/interop-2022-viewport#18]
This is a placeholder issue to discuss `-webkit-fill-available`. It is currently not covered in [the test scenarios](https://goo.gle/interop-2022-viewport-testresults) nor covered by [an explainer](https://github.com/web-platform-tests/interop-2022-viewport#definitions-and-findings).

**Comment by karlcow** [comment]
Do you mean covering `stretch` value in the tests?

`width`/`height`, `(min|max)-(width|height)` with the value `stretch`

* Firefox 
  * [Unprefix -moz-fit-content and -moz-available keywords](https://bugzilla.mozilla.org/show_bug.cgi?id=1495868). This is a Webcompat Priority P2 for Mozilla, as it creates webcompat issues. 
  * [support -moz-fit-content, and -moz-available for height/min-height/max-height (block axis)](https://bugzilla.mozilla.org/show_bug.cgi?id=567039)
  * [[css-sizing][meta] implement CSS Intrinsic & Extrinsic Sizing Module Level 3](https://bugzilla.mozilla.org/show_bug.cgi?id=1312587)
* Chromium
  * [Behavior of width:-webkit-fill-available](https://bugs.chromium.org/p/chromium/issues/detail?id=807708) Chromium
  * [[css-sizing] Unprefix fill/stretch-fit keyword when ready](https://bugs.chromium.org/p/chromium/issues/detail?id=611857)
* WebKit 
  * Nothing specific [but a list of related bugs](https://bugs.webkit.org/buglist.cgi?quicksearch=fill-available&list_id=8507743)
* CSS WG
  * [CSS Box Sizing Module Level 4: stretch old value](https://drafts.csswg.org/css-sizing-4/#valdef-width-stretch)


It's a mess. 

Two years ago I was wondering [if there was a counter for `-webkit-fill-available`](https://bugzilla.mozilla.org/show_bug.cgi?id=1495868#c7) @foolip

**Comment by bramus** [comment]
- @emilio CSS Keyword, was supposed to only work on the inline axis. WebKit does something on the block axis, which authors now rely on.
- @emilio We don’t want this implement this right now.
- @emilio Useful for some viewport resizing behavior
- @emilio We could probably ask @bfgeek about it
- @dcrousso Will ask around as well

**Comment by bfgeek** [comment]
We've been slowly moving our `-webkit-fill-available` behaviour to the behaviour of what `stretch` should be.

The TL;DR of `stretch` is that is should act similarly[1] to `100%` in standards mode. E.g. if the available-size in that dimension is definite.

(The way that I think about this at least is that `auto` maps onto either `fit-content` or `stretch` depending on the layout-mode and type of object. E.g. in flex in the cross-axis `auto` maps to `stretch` for example).

Our old implementation (and likely similar to current WebKit) behaved similarly to quirks-mode `100%`. E.g. it would skip past any `auto` height elements in the containing-block chain up to the first "definite" thing. As a result of this lots of folks relied on this behaviour as it kinda-sorta behaved like `100dvh` sometimes.

Blink can likely unprefix `-webit-fill-available` as `stretch` as we are reasonably confident that our `-webkit-fill-available` behaviour matches the spec.

Ian
